platform-service-impl/src/main/java/esa/mo/platform/impl/provider/opssat/PowerControlOPSSATAdapter.java

```diff
@@ -123,17 +145,66 @@ public void enableDevices(final DeviceList inputList) throws IOException
       }
       if (payloadId != null) {
         switchDevice(payloadId, device.getEnabled());
+        // When device is set OFF, mark the status right away
+        if(!device.getEnabled())
+        {
+          Device d = findByType(device.getDeviceType());
+          d.setEnabled(false);
+          if(d.getDeviceType() == DeviceType.ADCS){
+            adcsChannelStartTime = null;
+          }
+        }
       } else {
         throw new IOException("Cannot find the device by oId " + device);
       }
     }
   }
 
-  private void switchDevice(final PayloadDevice device, final Boolean enabled) throws IOException
+
+  @Override
+  public boolean isDeviceEnabled(DeviceType deviceType) {
+    boolean isEnabled = findByType(deviceType) == null ? false : findByType(deviceType).getEnabled();
+
+    // In the ADCS case, check that the power channel has been running at least a minute to get an accurate reading
+    if( deviceType.equals(DeviceType.ADCS) && (System.currentTimeMillis() - adcsChannelStartTime < 60*1000))
+    {
+       isEnabled = false;
+    }
+    return isEnabled;
+  }
+
+  @Override
+  public void startStatusTracking(ConnectionConsumer connection) {
+    CompleteDataReceivedListener listener = new CompleteDataReceivedListener() {
+      @Override
+      public void onDataReceived(ParameterInstance parameterInstance) {
```

Comment:
Ensure that the parameterInstance actually covers the parameter we are interested in `PDU1952`. Disregard it otherwise.

Reply:
Done. See the next line.